LEGO1/mxpresenter.cpp

```diff
@@ -1,26 +1,78 @@
 #include "mxpresenter.h"
 #include "mxautolocker.h"
+#include "mxparam.h"
+#include "legoomni.h"
+#include <string.h>
 
 #include "decomp.h"
 
 DECOMP_SIZE_ASSERT(MxPresenter, 0x40);
 
+// 0x10101eac
+char *g_parseExtraTokens = ":;";
+
+// 0x10101edc
+char *g_strWORLD = "WORLD";
+
 // OFFSET: LEGO1 0x100b4d50
 void MxPresenter::Init()
 {
   m_currentTickleState = TickleState_Idle;
   m_action = NULL;
   m_unk0x18 = 0;
-  m_unk0x3c = 0;
+  m_unkPresenter = NULL;
   m_previousTickleStates = 0;
   m_unk0x10 = 0;
   m_unk0x14 = 0;
 }
 
-// OFFSET: LEGO1 0x100b4fc0 STUB
+// OFFSET: LEGO1 0x100b4fc0
 void MxPresenter::ParseExtra()
 {
-  // TODO
+
+  MxAutoLocker lock(&m_criticalSection);
+  MxU32 len = m_action->GetUnkLength();
+  char *unk_data = m_action->GetUnkData();
+
+  if (len) {
+    len &= MAXWORD;
+    char t_actionData[512];
+    memcpy(t_actionData, unk_data, len);
+    t_actionData[len] = '\0';
+
+    char t_worldValue[512];
+    if (KeyValueStringParse(t_worldValue, g_strWORLD, t_actionData)) {
+      char *token = strtok(t_worldValue, g_parseExtraTokens);
+      char t_token[256];
+      strcpy(t_token, token);
+
+      token = strtok(NULL, g_parseExtraTokens);
+      int val = token ? atoi(token) : 0;
+
+      int result = MxOmni::GetInstance()->vtable0x30(t_token, val, this);
+      
+      // TODO: magic number for flag
+      m_action->SetFlags(m_action->GetFlags() | 128);
```

Comment:
Maybe `128` can be added to the flags enum in `MxDSAction`? It's still unknown but we could call it `Flag_unk128` or something

Reply:
Hex is probably preferable too for `or` flags like these

Reply:
I took a leap here and named the flag "Parsed" since we are setting it in the ParseExtra function